genson/src/test/java/com/owlike/genson/convert/JsonDeserializationTest.java

```diff
@@ -357,6 +357,47 @@ public void testDeserializeEnum() {
     assertEquals("foo", pojo.str);
   }
 
+  @Test
+  public void testDeserializationOptional() {
```

Comment:
You don't really need `@class` for any of these, as you are providing type to deserialize into explicitly.